### PR TITLE
fix: replace unix epoch to time.Time in Transaction

### LINKED ISSUE #27: Inconsistent API: timestamp vs time.Time
At the moment library handles time in two different ways: 
- [as time.Time](https://github.com/shal/mono/blob/master/personal.go#L42)
- [as int32 timestamp](https://github.com/shal/mono/blob/master/types.go#L50)
 
It's not a huge deal, but it's a little bit inconvenient in situations, when user of the library wants to use latest synced transaction time as an argument for the next `.Transactions()` call: user would need to convert timestamp to time.Time and it would be converted back to timestamp a few rows later.

I suggest to use single approach here (doesn't care which one, but personally I would prefer raw timestamp - because mono api uses it internally and we would not introduce any abstraction over it for the end user).
But, if we go with time.Time - we would need to use [something like this](https://www.yellowduck.be/posts/handling-unix-timestamps-in-json/) for time parsing in mono api responses.

@shal  what do you think?
I could implement a PR, just say what road are we going here: time.Time vs raw timestamp

### PR DESCRIPTION
**Pull request by SuddenGunter**

### CODE REVIEW

File: time_test.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
+package mono
+
+import (
+	"encoding/json"
+	"testing"
+	"time"
+)
+
+func TestTime_JsonUnmarshal(t *testing.T) {
```
Comment (shal) [inline time_test.go]: Usually, we call tests with the following name convention: `Test<Type>_<Method>`.
Let's call this test `TestTime_ UnmarshalJSON`.

File: time_test.go [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+	jsonString := []byte(`
```
Comment (shal) [inline time_test.go]: Something happened here, could you please replace it to ```[]byte(`{"time": 1583000643}`)```.

File: time_test.go [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    {
+        "time": 1583000643
+	}`)
+	expectedData := Transaction{
```
Comment (shal) [inline time_test.go]: I don't think, that test should actually now about `Transaction`, let's do simple `type struct{Time Time}` for testing.

File: time_test.go [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+	if err != nil {
+		t.Fatalf("expected error: nil, actual error: %v", err)
+	}
+	if expectedData != actualData {
+		t.Fatalf("expected data: %v, actual data: %v", expectedData, actualData)
+	}
+}
+
+func TestTime_JsonMarshal(t *testing.T) {
```
Comment (shal) [inline time_test.go]: Let's call this test `TestTime_ MarshalJSON `.

File: personal_test.go [hunk 5]
```diff
@@ -40,23 +43,80 @@ func TestPersonal_User(t *testing.T) {
...
-//			Amount:          -95000,
-//			OperationAmount: -95000,
-//			CurrencyCode:    980,
-//			CommissionRate:  0,
-//			CashBackAmount:  19000,
-//		},
-//	}
-//
-//}
+func TestPersonal_Transactions(t *testing.T) {
```
Comment (shal) [inline personal_test.go]: Unfortunately, we shouldn't put this changes into the PR. It is out of the scope. 
Our tests are pretty poor, let's keep these changes for PR for #29.

File: time_test.go [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+	}`)
+	expectedData := Transaction{
+		Time: Time{time.Date(2020, 2, 29, 18, 24, 03, 00, time.UTC)},
+	}
+
+	var actualData Transaction
+	err := json.Unmarshal(jsonString, &actualData)
+
+	if err != nil {
+		t.Fatalf("expected error: nil, actual error: %v", err)
```
Comment (shal) [inline time_test.go]: `t.Errorf` would be a better solution here.

File: time_test.go [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+	}
+	if expectedData != actualData {
```
Comment (shal) [inline time_test.go]: There is a helper for that: `assertEqual`. Would you mind using it?

**Review by shal — CHANGES_REQUESTED** [review]

**Comment by SuddenGunter** [comment]
@shal fixed

**Review by shal — APPROVED** [review]

**Pull request merged**